src/core/qgsproject.h

```diff
@@ -121,6 +121,7 @@ class CORE_EXPORT QgsProject : public QObject, public QgsExpressionContextGenera
       FlagDontResolveLayers = 1 << 0, //!< Don't resolve layer paths (i.e. don't load any layer content). Dramatically improves project read time if the actual data from the layers is not required.
       FlagDontLoadLayouts = 1 << 1, //!< Don't load print layouts. Improves project read time if layouts are not required, and allows projects to be safely read in background threads (since print layouts are not thread safe).
       FlagTrustLayerMetadata = 1 << 2, //!< Trust layer metadata. Improves project read time. Do not use it if layers' extent is not fixed during the project's use by QGIS and QGIS Server.
+      FlagDontStoreOriginalStyles = 1 << 3, //!< Skip the initial XML style storage for layers. Mainly useful in server's context where there is no chance that the user will save the project again.
```

Comment:
Actually there's some other good use cases for this, eg when expanding out projects in the browser panel and we just want the projects to open as fast as possible.

Maybe the docs could reflect this by saying something like 'usefulnfor minimising project load times in non-interactive contexts"